contracts/lending/interfaces/IAddressProvider.sol

```diff
@@ -0,0 +1,15 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.19;
+
+interface IAddressProvider {
+    event NFTfiLendingUpdated(address indexed newAddress);
+    event X2Y2LendingUpdated(address indexed newAddress);
+
+    function getNFTfiLending() external view returns (address);
+
+    function setNFTfiLending(address _nftfiLending) external;
+
+    function getX2Y2Lending() external view returns (address);
+
+    function setX2Y2Lending(address _x2y2Lending) external;
```

Comment:
I don't think we need custom function for each lending, we would pass in a contract address or enum in the call which will determine which protocol to use